src/core/Cline.ts

```diff
@@ -1037,12 +1037,16 @@ export class Cline {
 							if (
 								newContent.includes("&gt;") ||
 								newContent.includes("&lt;") ||
-								newContent.includes("&quot;")
+								newContent.includes("&quot;") ||
+								newContent.includes("&amp;") ||
+								newContent.includes("&apos;")
```

Comment:
can be simplified to 
`newContent.match(/&\w+;/g)` 
or if you want to be specific 
```js
function isUnallowedHtmlEntities(content) {
   const entityNamesToBeBypassed = ['gt','lt','quot','amp','apos'] // can be extended later
   const unallowedEntityNamesRegExp = entityNamesToBeBypassed.map(entityName => `(${entityName})`).join('|')
   const reg = new RegExp(`&${unallowedEntityNamesRegExp};`,'g')
   return content.match(reg).length > 0
}

// Usage
if(isUnallowedHtmlEntities(content) {
...
}
```